traffic_ops/traffic_ops_golang/types/types.go

```diff
@@ -95,9 +98,48 @@ func (typ *TOType) Validate() error {
 }
 
 func (tp *TOType) Read() ([]interface{}, error, error, int) { return api.GenericRead(tp) }
-func (tp *TOType) Update() (error, error, int)              { return api.GenericUpdate(tp) }
-func (tp *TOType) Create() (error, error, int)              { return api.GenericCreate(tp) }
-func (tp *TOType) Delete() (error, error, int)              { return api.GenericDelete(tp) }
+
+func (tp *TOType) Update() (error, error, int) {
+	if !tp.AllowMutation() {
+		return nil, errors.New("can not update type"), http.StatusBadRequest
+	}
+	return api.GenericUpdate(tp)
+}
+
+func (tp *TOType) Delete() (error, error, int) {
+	if tp.UseInTable == nil {
```

Comment:
when I try to delete an ID that doesnt exist I am getting 

```
ERROR: api.go:789: 2020-03-12T18:53:41.8901632Z: a non-pq error was given
ERROR: api.go:173: 2020-03-12T18:53:41.8916858Z: 172.28.0.1:60308 sql: no rows in result set
DEBUG: api.go:178: 2020-03-12T18:53:41.8918061Z: Internal Server Error
```

are you getting the same?

Also wont `tp.UseInTable == nil` always be nil? since the object isnt passed in on request?

Reply:
You are correct, at the time of writing i was confused as to how the generic CRUD was working. I'll remove it in my upcoming changes.